mongo/mongo.go

```diff
@@ -0,0 +1,362 @@
+//go:build !nomongo
+// +build !nomongo
+
+package mongo
+
+import (
+	"bytes"
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/nosql/database"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/writeconcern"
+)
+
+// DB is a wrapper over *sql.DB,
+type DB struct {
+	db *mongo.Database
+}
+
+type Tuple struct {
+	Key   []byte `bson:"key" json:"key"`
+	Value []byte `bson:"value" json:"value"`
+}
+
+// Open creates a Driver and connects to the database with the given address
+// and access details.
+func (db *DB) Open(uri string, opt ...database.Option) error {
+	opts := &database.Options{}
+	for _, o := range opt {
+		if err := o(opts); err != nil {
+			return err
+		}
+	}
+
+	clientOptions := options.Client().ApplyURI(uri)
+	if rs := clientOptions.ReplicaSet; *rs == "" {
+		return errors.New("To enable transactions, please provide a replica set name")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	client, err := mongo.Connect(ctx, clientOptions)
+	if err != nil {
+		return errors.Wrap(err, "error in configuration")
+	}
+
+	err = client.Ping(context.Background(), nil)
+	if err != nil {
+		return errors.Wrap(err, "error connecting to mongo")
+	}
+
+	dbOpts := options.Database()
+	db.db = client.Database(opts.Database, dbOpts)
+
+	return nil
+}
+
+func (db *DB) Close() error {
+	if err := db.db.Client().Disconnect(context.Background()); err != nil {
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func (db *DB) CreateTable(bucket []byte) error {
```

Comment:
Wherever you use `bucket` here, you can rename it to `collection`, if it makes sense to do so.

Reply:
What's the reasoning? Not that I disagree but is this a mongo specific thing or would given the chance you also update the use of bucket in the other adapters too?